src/store-api/src/region_request.rs

```diff
@@ -446,6 +463,39 @@ impl AlterKind {
         );
         Ok(())
     }
+
+    pub fn validate_column_fulltext_to_alter(
+        metadata: &RegionMetadata,
+        column_name: &str,
+        options: &HashMap<String, String>,
+    ) -> Result<()> {
+        let _ =
+            metadata
+                .column_by_name(column_name)
+                .with_context(|| InvalidRegionRequestSnafu {
+                    region_id: metadata.region_id,
+                    err: format!("column {} not found", column_name),
+                })?;
+
+        for (k, v) in options {
+            if k.to_ascii_lowercase().eq(COLUMN_FULLTEXT_OPT_KEY_ANALYZER) {
+                if v.to_ascii_lowercase().ne("english") && v.to_ascii_lowercase().ne("chinese") {
+                    return InvalidFulltextOptionsSnafu { column_name }.fail();
+                }
+            } else if k
+                .to_ascii_lowercase()
+                .eq(COLUMN_FULLTEXT_OPT_KEY_CASE_SENSITIVE)
+            {
+                if v.to_ascii_lowercase().ne("true") && v.to_ascii_lowercase().ne("false") {
+                    return InvalidFulltextOptionsSnafu { column_name }.fail();
+                }
+            } else {
+                return InvalidFulltextOptionsSnafu { column_name }.fail();
+            }
+        }
+
+        Ok(())
+    }
```

Comment:
Let's introduce a method to validate the options, which you can reuse in other places. And remember to add corresponding unit tests.